test-support/helpers/mirage-unit-helpers.js

```diff
@@ -0,0 +1,71 @@
+import Ember from 'ember';
+
+let sequenceMap = {};
+
+var store;
+var setStore = function(passedStore) {
+  store = passedStore;
+};
+
+var _factoryCreate = function(factory, modelName) {
+  let obj = _factoryBuild(factory, modelName);
+  let record;
+
+  Ember.run(function(){
+    let data = _.defaults(obj, {id: sequenceMap[modelName]});
+    record = store.push(
+      modelName,
+      data
+    );
+  });
+  return record;
+
+};
+
+var _factoryBuild = function(factory, modelName) {
+  if (!sequenceMap.hasOwnProperty(modelName)) {
+    sequenceMap[modelName] = 1;
+  }
+  return factory.build(sequenceMap[modelName]++);
+};
+
+var _getFactory = function(Factory, overrides={}) {
+  let FactoryWithOverrides = Factory.extend(overrides);
+  return new FactoryWithOverrides();
+};
+
+var build = function(Factory, modelName, overrides={}) {
```

Comment:
Since `Factory` has no knowledge of the model type the name needs to be provided.  I'm very open to suggestions for improvement here.